dm/dm/master/shardddl/optimist.go

```diff
@@ -135,31 +140,159 @@ func (o *Optimist) ShowLocks(task string, sources []string) []*pb.DDLLock {
 			continue // specify sources but mismath
 		}
 	FOUND:
-		l := &pb.DDLLock{
-			ID:       lock.ID,
-			Task:     lock.Task,
-			Mode:     config.ShardOptimistic,
-			Owner:    "",  // N/A for the optimistic mode
-			DDLs:     nil, // N/A for the optimistic mode
-			Synced:   make([]string, 0, len(ready)),
-			Unsynced: make([]string, 0, len(ready)),
+		var (
+			owners    []string
+			ddlGroups [][]string
+		)
+
+		appendOwnerDDLs := func(opmss map[string]map[string]optimism.Operation, source string) {
```

Comment:
closure in for loop looks so confused.

Reply:
addressed in https://github.com/pingcap/tiflow/pull/4087/commits/96b2db76737e4e6221b5f0413690e0be2d8a5e15

Reply:
It doesn't seem to have any change?

Reply:
I have changed if part to continue. Do you mean the variable names are confused? Or do you some other suggestions?